source/Calamari.Common/Features/Scripting/ScriptEngine.cs

```diff
@@ -104,12 +108,21 @@ IScriptExecutor GetScriptExecutor(ScriptSyntax scriptSyntax, bool runDotnetScrip
                 case ScriptSyntax.Bash:
                     return new BashScriptExecutor();
                 case ScriptSyntax.FSharp:
+                    LogFSharpDeprecationWarning(variables);
                     return new FSharpExecutor();
                 case ScriptSyntax.Python:
                     return new PythonScriptExecutor();
                 default:
                     throw new NotSupportedException($"{scriptSyntax} script are not supported for execution");
             }
         }
+
+        void LogFSharpDeprecationWarning(IVariables variables)
+        {
+            if (FeatureToggle.FSharpDeprecationFeatureToggle.IsEnabled(variables))
+            {
+                log.Warn($"Executing FSharp scripts will soon be deprecated. Please read our deprecation {log.FormatLink("https://octopus.com/blog/2024-deprecated-features#f-sharp", "blog post")} for more details.");
```

Comment:
This url should be converted to use a short URL
https://octopushq.atlassian.net/wiki/spaces/MAR/pages/2393899875/Short+URLs+process+and+rules
This ensures that if we move the underlying page in future these links continue to work